src/main/java/com/iexec/worker/executor/TaskNotificationService.java

```diff
@@ -0,0 +1,189 @@
+                break;
+            case PLEASE_DOWNLOAD_DATA:
+                updateStatusAndGetNextAction(chainTaskId, DATA_DOWNLOADING);
+                boolean isDataDownloaded = taskManagerService.downloadData(chainTaskId);
+                if (!isDataDownloaded) {
+                    updateStatusAndGetNextAction(chainTaskId, DATA_DOWNLOAD_FAILED);
+                    return;
+                }
+                nextAction = updateStatusAndGetNextAction(chainTaskId, DATA_DOWNLOADED);
+                break;
+            case PLEASE_COMPUTE:
+                updateStatusAndGetNextAction(chainTaskId, COMPUTING);
+                boolean isComputed = taskManagerService.compute(chainTaskId);
+                if (!isComputed) {
+                    updateStatusAndGetNextAction(chainTaskId, COMPUTE_FAILED);
+                    return;
+                }
+                nextAction = updateStatusAndGetNextAction(chainTaskId, COMPUTED);
+                break;
+            case PLEASE_CONTRIBUTE:
+                updateStatusAndGetNextAction(chainTaskId, CONTRIBUTING);
+                boolean isContributed = taskManagerService.contribute(chainTaskId);
+                if (!isContributed) {
+                    updateStatusAndGetNextAction(chainTaskId, CONTRIBUTE_FAILED);
+                    return;
+                }
+                nextAction = updateStatusAndGetNextAction(chainTaskId, CONTRIBUTED);
+                break;
+            case PLEASE_REVEAL:
+                updateStatusAndGetNextAction(chainTaskId, REVEALING);
+                boolean isRevealed = taskManagerService.reveal(chainTaskId, extra);
+                if (!isRevealed) {
+                    updateStatusAndGetNextAction(chainTaskId, REVEAL_FAILED);
+                    return;
+                }
+                nextAction = updateStatusAndGetNextAction(chainTaskId, REVEALED);
+                break;
+            case PLEASE_UPLOAD:
+                updateStatusAndGetNextAction(chainTaskId, RESULT_UPLOADING);
+                ReplicateDetails uploadDetails = taskManagerService.uploadResult(chainTaskId);
+                if (uploadDetails == null) {
+                    updateStatusAndGetNextAction(chainTaskId, RESULT_UPLOAD_FAILED);
+                    return;
+                }
+                nextAction = updateStatusAndGetNextAction(chainTaskId, RESULT_UPLOADED, uploadDetails);
+                break;
+            case PLEASE_COMPLETE:
+                updateStatusAndGetNextAction(chainTaskId, COMPLETING);
+                boolean isCompleted = taskManagerService.complete(chainTaskId);
+                if (!isCompleted) {
+                    updateStatusAndGetNextAction(chainTaskId, COMPLETE_FAILED);
+                    return;
+                }
+                updateStatusAndGetNextAction(chainTaskId, COMPLETED);
+                break;
+            //TODO merge abort
+            case PLEASE_ABORT_CONTRIBUTION_TIMEOUT:
```

Comment:
PLEASE_ABORT_CONTRIBUTION_TIMEOUT and PLEASE_ABORT_CONSENSUS_REACHED are using the same code, they can be merged using:
[...]
case PLEASE_ABORT_CONTRIBUTION_TIMEOUT:
case PLEASE_ABORT_CONSENSUS_REACHED:
[...]